src/crawler/crawl.ts

```diff
@@ -233,7 +233,7 @@ class Crawler {
       }
 
       this.publicKeysSeen.add(normalizedPublicKey)
```

Comment:
Shouldn't this be gated by an `if` condition? How do you ensure that you are not over-writing the valid ip/port endpoints with the NULL values? (I'm referring to those server operators who still want to be discovered by the community)

Reply:
Oh wait, since you have defined these fields `ip: undefined, port: undefined` in the beginning of the object, they can be over-written with valid values, if they exist. that makes sense